app/models/finance/postpaid_billing_strategy.rb

```diff
@@ -35,13 +35,19 @@ def description
   end
 
   def bill_plan_change(contract, period)
-    if contract.paid_until.to_date < period.begin.to_date
-      add_plan_cost(:bill, contract, contract.plan, period)
-    else
-      # FIXME: replace end_of_month by paid until!
-      add_plan_cost(:refund, contract, contract.old_plan, period)
-      add_plan_cost(:bill, contract, contract.plan, period)
+    plan = contract.plan
+    old_plan = contract.old_plan
+    paid_until = contract.paid_until.utc
+
+    if paid_until < period.begin.utc
+      # This can possibly cross several months
+      for_every_month_within(TimeRange.new(paid_until, period.end.utc)) do |month|
```

Comment:
It should probably be the intersection and therefore no need for `for_every_month_within`.